src/Microsoft.Extensions.Configuration.FileProviderExtensions/ConfigurationRootExtensions.cs

```diff
@@ -20,8 +21,13 @@ public static IConfigurationRoot ReloadOnChanged(this IConfigurationRoot config,
                 throw new ArgumentNullException(nameof(filename));
             }
 #if NET451
-            var basePath = AppDomain.CurrentDomain.GetData("APP_CONTEXT_BASE_DIRECTORY") as string ??
-                AppDomain.CurrentDomain.BaseDirectory ?? string.Empty;
+            var basePath =
```

Comment:
`GetBasePath` \ `SetBasePath` are on `IConfigurationBuilder`. `IConfigurationRoot` does not have access to it. Any suggestions on what we should do here?